src/components/alert/Alert.tsx

```diff
@@ -1,27 +1,40 @@
 import React from 'react';
 import { DialogBase } from '../dialog';
-import PropTypes from 'prop-types';
 import { Heading3, TextBody } from '../typography';
 import { Box } from '../box';
 import { Button, BUTTON_LEVELS } from '../button';
 import omit from 'lodash.omit';
 import { Illustration60X60Error, Illustration60X60Info } from '@teamleader/ui-illustrations';
+import { DialogBaseProps } from '../dialog/DialogBase';
 
-const ALERT_TYPES = {
-  confirm: 'confirm',
-  destructive: 'destructive',
-  error: 'error',
-};
+interface AlertProps extends DialogBaseProps {
+  /** Object containing the props of the primary action (a Button, with level prop set to 'primary'). */
+  primaryAction: Action;
+  /** Object containing the the props of the secondary action (a Button). */
+  secondaryAction?: Action;
+  /** The title of the alert. */
+  title: string;
+  /** The body of the alert. */
+  body?: string;
+  /** The type of the alert. */
+  type?: ALERT_TYPES;
+}
+
+interface Action {
+  label: string;
+  onClick: () => void;
+}
 
-/** @type {React.ComponentType<any>} */
-const Alert = ({ primaryAction, secondaryAction, title, body, type, ...otherProps }) => {
+type ALERT_TYPES = 'confirm' | 'destructive' | 'error';
+
+const Alert = ({ primaryAction, secondaryAction, title, body, type = 'confirm', ...otherProps }: AlertProps) => {
   const restProps = omit(otherProps, ['primaryAction', 'secondaryAction']);
```

Comment:
The omit is no longer necessary because both `primaryAction` and `secondaryAction` aren't a rest prop anymore (see previous line)